### PR TITLE
[Fix] `order`: restore default behaviour unless `type` is in groups

### PR DESCRIPTION
**Pull request by grit96**
#2070 and #2084

### CODE REVIEW

File: tests/src/rules/order.js [hunk 1]
```diff
@@ -2258,14 +2302,14 @@ context('TypeScript', function () {
               parser,
               options: [
                 {
-                  groups: ['external', 'index'],
+                  groups: ['external', 'index', 'type'],
```
Comment (ljharb) [inline tests/src/rules/order.js:2305]: confirming: this was the test you added, thus, the only impact is on the feature from that PR?

File: tests/src/rules/order.js [hunk 2]
```diff
@@ -2281,7 +2325,7 @@ context('TypeScript', function () {
...
```
Comment (ljharb) [inline tests/src/rules/order.js:2328]: same here?

File: tests/src/rules/order.js [hunk 3]
```diff
@@ -2303,35 +2347,130 @@ context('TypeScript', function () {
...
                 {
                   groups: ['external', 'index'],
+                  alphabetize: { order: 'desc' },
+                },
+              ],
+              errors: [
+                {
+                  message: semver.satisfies(eslintPkg.version, '< 3')
+                    ? '`bar` import should occur before import of `Bar`'
+                    : /(`bar` import should occur before import of `Bar`)|(`Bar` import should occur after import of `bar`)/,
```
Comment (ljharb) [inline tests/src/rules/order.js:2441]: why both, in eslint >= 4?

**Review by ljharb — APPROVED** [review]
Can you confirm that all changed tests were ones you added in your original PR?

**Comment by grit96** [comment]
@ljharb yes, I reverted the test changes from the previous commit and added new ones for the type group

File: tests/src/rules/order.js (hunk 2 above)
Comment (grit96) [reply tests/src/rules/order.js:2328]: yes, I reverted the test changes from the previous commit and added new ones for the type group

File: tests/src/rules/order.js (hunk 1 above)
Comment (grit96) [reply tests/src/rules/order.js:2305]: yes, I reverted the test changes from the previous commit and added new ones for the type group

File: tests/src/rules/order.js (hunk 3 above)
Comment (grit96) [reply tests/src/rules/order.js:2441]: I think I just copied this from the other tests, I can change it if required.

**Comment by mmiszy** [comment]
Could you, please, stop introducing breaking changes in `patch` releases? Thank you :P

**Comment by ljharb** [comment]
@mmiszy the word you’re looking for is “bugs”. Feel free to request a refund though.

**Comment by mmiszy** [comment]
@ljharb there's no need to be sarcastic and condescending.

**Comment by ljharb** [comment]
@mmiszy yes, i agree that your initial comment was both rude and unnecessary.

**Comment by mmiszy** [comment]
@ljharb I'm sorry that you feel that way about my feedback.

**Comment by ljharb** [comment]
@mmiszy it's not feedback. nobody intentionally introduces breaking changes in a patch release - it's a bug that it's happened. Bugs are unavoidable, and can never be fully stopped, so asking for that can't possibly ever achieve anything except to make someone feel bad that they introduced a bug (which in this case, wasn't even me).

Please stop pretending like your comment was made in good faith. It was clearly snark born out of frustration, and that should be kept to oneself.

**Comment by mmiszy** [comment]
@ljharb I'm sorry you see it this way 😥

**Pull request merged**